src/games/strategy/triplea/ai/proAI/data/ProMoveOptions.java

```diff
@@ -69,48 +63,52 @@ private void populateMaxMoveMap(final List<Map<Territory, ProTerritory>> moveMap
         }
 
         // Check if mover's turn comes before territory owner's
-        if (!utils.isPlayersTurnFirst(players, movePlayer, t.getOwner())) {
+        if (!ProUtils.isPlayersTurnFirst(players, movePlayer, t.getOwner())) {
           continue;
         }
 
         // Add to max move map if its empty or its strength is greater than existing
-        if (!maxMoveMap.containsKey(t)) {
-          maxMoveMap.put(t, moveMap.get(t));
+        if (!result.containsKey(t)) {
+          result.put(t, moveMap.get(t));
         } else {
-          final Set<Unit> maxUnits = new HashSet<Unit>(maxMoveMap.get(t).getMaxUnits());
-          maxUnits.addAll(maxMoveMap.get(t).getMaxAmphibUnits());
+          final Set<Unit> maxUnits = new HashSet<Unit>(result.get(t).getMaxUnits());
+          maxUnits.addAll(result.get(t).getMaxAmphibUnits());
           double maxStrength = 0;
           if (!maxUnits.isEmpty()) {
             maxStrength =
-                battleUtils.estimateStrength(maxUnits.iterator().next().getOwner(), t, new ArrayList<Unit>(maxUnits),
-                    new ArrayList<Unit>(), isAttacker);
+                ProBattleUtils.estimateStrength(maxUnits.iterator().next().getOwner(), t,
+                    new ArrayList<Unit>(maxUnits), new ArrayList<Unit>(), isAttacker);
           }
           final double currentStrength =
-              battleUtils.estimateStrength(currentUnits.iterator().next().getOwner(), t, new ArrayList<Unit>(
+              ProBattleUtils.estimateStrength(currentUnits.iterator().next().getOwner(), t, new ArrayList<Unit>(
                   currentUnits), new ArrayList<Unit>(), isAttacker);
           final boolean currentHasLandUnits = Match.someMatch(currentUnits, Matches.UnitIsLand);
           final boolean maxHasLandUnits = Match.someMatch(maxUnits, Matches.UnitIsLand);
           if ((currentHasLandUnits && ((!maxHasLandUnits && !t.isWater()) || currentStrength > maxStrength))
               || ((!maxHasLandUnits || t.isWater()) && currentStrength > maxStrength)) {
-            maxMoveMap.put(t, moveMap.get(t));
+            result.put(t, moveMap.get(t));
           }
         }
       }
     }
+    return result;
   }
 
-  private void populateMoveMaps(final List<Map<Territory, ProTerritory>> moveMapList) {
+  private static Map<Territory, List<ProTerritory>> createMoveMaps(final List<Map<Territory, ProTerritory>> moveMapList) {
+
+    final Map<Territory, List<ProTerritory>> result = new HashMap<Territory, List<ProTerritory>>();
     for (final Map<Territory, ProTerritory> moveMap : moveMapList) {
       for (final Territory t : moveMap.keySet()) {
-        if (!moveMaps.containsKey(t)) {
+        if (!result.containsKey(t)) {
           final List<ProTerritory> list = new ArrayList<ProTerritory>();
           list.add(moveMap.get(t));
-          moveMaps.put(t, list);
+          result.put(t, list);
```

Comment:
Code smells:
- Deeply nested code
- Complicated use of simple data structures. Promote these Maps and other data structures to first class objects so you can break up and hide some of the complexity in those classes.